app/services/pdf_processor.py

```diff
@@ -0,0 +1,179 @@
+import fitz
+from transformers import pipeline
+from fuzzywuzzy import fuzz
+import json
+
+
+def extract_text_and_coordinates_by_line(pdf_path):
+    """
+    Extracts text and coordinates from the given PDF by line using PyMuPDF.
+
+    Parameters:
+    pdf_path (str): The path to the PDF file.
+
+    Returns:
+    list: A list of dictionaries, each containing the line text and its coordinates.
+    """
+    document = fitz.open(pdf_path)
+    lines_with_coords = []
+
+    for page_num in range(len(document)):
+        page = document[page_num]
+        blocks = page.get_text("dict")["blocks"]
+        for block in blocks:
+            if "lines" in block:
+                for line in block["lines"]:
+                    line_text = ""
+                    line_coords = []
+                    for span in line["spans"]:
+                        line_text += span["text"] + " "
+                        line_coords.append((span["bbox"], span["text"]))
+                    lines_with_coords.append({
+                        "text": line_text.strip(),
+                        "coords": line_coords,
+                        "page": page_num
+                    })
+
+    return lines_with_coords
+
+
+def extract_names(text):
+    """
+    Extracts names from the given text using a Named Entity Recognition (NER) model.
+
+    Parameters:
+    text (str): The input text from which to extract names.
+
+    Returns:
+    list:  A list of extracted names.
+    """
+    ner_pipeline = pipeline("ner",
```

Comment:
You need to make sure that the model is not downloaded each time. Sometimes they always download something and you need to create a model cache to avoid delays in the response.